ir/type.def

```diff
@@ -233,6 +233,15 @@ abstract Type_StructLike : Type_Declaration, ISimpleNamespace, IAnnotated {
         return fields.getDeclarations(); }
     StructField getField(cstring name) const {
         return fields.getDeclaration<StructField>(name); }
+    int getFieldIndex(cstring name) const {
+        int index_pos = 0;
+        for (auto f : fields) {
+            if (f->name == name)
+                return index_pos;
+	    index_pos++;
+        }
+        return index_pos;
```

Comment:
How about returning -1 for an error?
Otherwise this could be confused with a regular index.

Reply:
Agree.  However, I see that lot of test P4 programs uses data struct or header with zero fields.  So many Travis tests failed with the new error.  I am looking code for final fix.